spectree/plugins/falcon_plugin.py

```diff
@@ -229,7 +229,7 @@ def validate(
         if resp and resp.has_model():
             model = resp.find_model(_resp.status[:3])
             if model and isinstance(_resp.media, model):
-                _resp.media = _resp.media.dict()
+                _resp.media = _resp.media.json().encode('UTF-8')
```

Comment:
I guess you're looking for https://falcon.readthedocs.io/en/stable/api/media.html#supported-handler-types.

`resp.media` should be a Python type by default. It will be serialized later.

Reply:
Thanks for the hint. It can solve this particular problem, but I think that the problem itself is a bit bigger. With pydantic we can define custom serializers per model, ex datetime in modelA can be serialized as a timestamp, while in modelB as a string. If we first serialize model to dict, and then we transform that dict into JSON, we are losing the possibility to exercise pydantic possibilities.

You are right that `_resp.media` is a wrong place to put the string. What are your thoughts on putting it into `_resp.data`?

Reply:
You can write the serialized data to `resp.text`(str) or `resp.data`(byte) directly.

I don't know when should the same data use different serialization methods. From my understanding, it should keep the same.

BTW, spectree only returns what is returned from the endpoint function. It has nothing to do with the serialization part. Although you may change some behavior through the `after` [handle](https://github.com/0b01001001/spectree/blob/7a635e5dcf33e423561ea4e51ea32d5c51be943d/tests/test_plugin_flask.py#L32-L33).

Reply:
Thanks for your answers. Indeed it looks like im using a wrong method. 

I observed an issue when I try to use serialized data

```python
class UserProfile:
    @spec.validate(
        resp=Response(HTTP_200=Message, HTTP_403=None), tags=["api"]
    )
    def on_post(self, req, resp):
        resp.data = Message(text='it works', when=datetime.now()).json().encode('UTF-8')
```

In the body I get what i expect. Unfortunetely the response code is 500 ` 500 Internal Service Response Validation Error`.

Looks like spectree falcon plugin is trying to to parse the response even if `media` is no there
```
Traceback (most recent call last):
  File "pydantic/main.py", line 522, in pydantic.main.BaseModel.parse_obj
TypeError: 'NoneType' object is not iterable

```

https://github.com/0b01001001/spectree/blob/master/spectree/plugins/falcon_plugin.py#L237

Would it make sense to check the presence of `media` and parse only if its present?

Reply:
Yes. That'll be great.

Reply:
Great, so ill close the PR and open a new one with the check, later today or tomorrow morning